stdlib/string.gr

```diff
@@ -382,6 +382,32 @@ export let rec implode = (arr: Array<Char>) => {
   ret
 }
 
+/**
+ * Create a string that is the given string reversed.
+ *
+ * @param string: The string to reverse
+ * @returns A string whose characters are in the reverse order of the given string
+ *
+ * @example String.reverse("olleH") == "Hello"
+ *
+ * @since v0.5.0
+ */
+export let reverse = (string) => {
+  let len = wasmSafeLength(string)
```

Comment:
Since this isn't a disableGC context, it's very bad to use this function. I believe it should be reporting a warning.
```suggestion
  let len = length(string)
```

Reply:
@ospencer is looking into a constant time lookup with the array length instead